qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1517A.py

```diff
@@ -284,3 +284,17 @@ def use_high_resolution_adc(self) -> None:
                    .aad(chnum=self.channels[0],
                         adc_type=AAD.Type.HIGH_RESOLUTION)
                    .message)
+
+    def set_adc_type_mode_and_coeff(self, *,
```

Comment:
this is a duplicate of [private method `_setup_integration_time` from `B1500` class ("mainframe")](https://github.com/QCoDeS/Qcodes/blob/99f543dcd671b372a563bd6e8df938e18c3f02fb/qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1500.py#L138-L145). Hence:
1. it's only necessary to make it public (either by wrapping it in a new function or by just renaming it from "private" to "public")
2. since this function does not set anything that is directly(!) specific to SMU, it should not live in the SMU module driver. as per design of the actual instrument, one assigns particular ADC to a SMU, which establishes the linkage between settings of that ADC and that SMU. it does seem convenient to have this method in SMU module class, but it'm not comfortable with complicating the perception of an already complicated instrument.